File: Sorting/Sorting02.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";

        String[] stringTypeNumbers = new String[numbers.length];

        for(int i = 0; i < numbers.length; i++) {
            stringTypeNumbers[i] = String.valueOf(numbers[i]);
        }

        Arrays.sort(stringTypeNumbers, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return (o2+o1).compareTo(o1+o2);
            }
        });

        if(stringTypeNumbers[0].equals("0")) {
            return "0";
        }

        for(int i = 0; i < stringTypeNumbers.length; i++) {
            answer += stringTypeNumbers[i];
        }

        return answer;
    }
}

File: Sorting/Sorting02.cpp
#include <string>
#include<iostream>
#include <vector>
#include<algorithm>
using namespace std;

bool compare(int, int);
string solution(vector<int> numbers) {
    string answer = "";
    int count=0;
    
    sort(numbers.begin(),numbers.end(),compare);

    for(int i=0,j=numbers.size();i<j;i++){
        if(numbers[i]==0)
            count++;
        answer+=to_string(numbers[i]);
    }
    if(count==numbers.size())
        answer="0";

    return answer;
}
bool compare(int previousNumber,int nextNumber){
    string previousToNext=to_string(previousNumber)+to_string(nextNumber);
    string nextToPrevious=to_string(nextNumber)+to_string(previousNumber);
    if(previousToNext.compare(nextToPrevious)>0){
         return true;
     }else{
         return false;
     }
}